idb/postgres_migrations.go

```diff
@@ -59,38 +64,35 @@ func (db *PostgresIndexerDb) runAvailableMigrations(migrationStateJson string) (
 		}
 	}
 
-	if state.NextMigration >= len(migrations) {
-		return nil
-	}
-	if anyBlockers(state.NextMigration) {
-		// run migrations synchronously
-		log.Print("Indexer startup will be delayed until database migration completes")
-		return db.runMigrationState(&state, true)
-	} else {
-		log.Print("database migrations running in background...")
-		go db.runMigrationState(&state, false)
-		return nil
-	}
-}
-
-func (db *PostgresIndexerDb) runMigrationState(state *MigrationState, blocking bool) (err error) {
+	// Make migration tasks
 	nextMigration := state.NextMigration
-	for true {
-		err = migrations[nextMigration].migrate(db, state)
-		if err != nil {
-			return fmt.Errorf("error in migration %d: %v", nextMigration, err)
-		}
+	tasks := make([]migration.Task, 0)
+	for nextMigration < len(migrations) {
+		tasks = append(tasks, migration.Task{
+			Handler: wrapPostgresHandler(migrations[nextMigration].migrate, db, &state),
+			MigrationId: nextMigration,
+			Description: migrations[nextMigration].description,
+		})
 		nextMigration++
-		if nextMigration >= len(migrations) {
-			return nil
-		}
-		state.NextMigration = nextMigration
-		if blocking && !anyBlockers(nextMigration) {
-			log.Print("database migrations will complete in background...")
-			go db.runMigrationState(state, false)
-			return nil
-		}
 	}
+
+	// Add a task to mark migrations as done instead of using a channel.
```

Comment:
I thought this was cool, but maybe it's too clever? Changing start to `func Start(onCompletion func() error)` would also be a nice way to trigger a final routine:
```
db.migration.Start(db.markMigrationsAsDone)
```